### PR TITLE
Revise consequences for breakend variants

### PR DESCRIPTION
**Pull request by nuno-agostinho**
[ENSVAR-2086](https://www.ebi.ac.uk/panda/jira/browse/ENSVAR-2086) and related with Ensembl/ensembl-vep#1399

- Adds breakend predicate statements
- If a breakend position is located within a feature, VEP returns `feature_truncation` as a consequence (together with other consequences that also make sense)

This is an example output for some BND variants:

```
gnomAD_v2_BND_1_4       1:251749-258100 <BND>   ENSG00000228463 ENST00000424587 Transcript      downstream_gene_variant -       -       -       -       -       -       IMPACT=MODIFIER;DISTANCE=4915;STRAND=-1
gnomAD_v2_BND_1_4       1:251749-258100 <BND>   ENSG00000228463 ENST00000441866 Transcript      downstream_gene_variant -       -       -       -       -       -       IMPACT=MODIFIER;DISTANCE=44;STRAND=-1
gnomAD_v2_BND_1_4       1:251749-258100 <BND>   ENSG00000228463 ENST00000442116 Transcript      feature_truncation,non_coding_transcript_exon_variant   1056-?  -       -       -       -       -       IMPACT=HIGH;STRAND=-1;OverlapBP=237;OverlapPC=3.45
gnomAD_v2_BND_1_4       1:251749-258100 <BND>   ENSG00000228463 ENST00000448958 Transcript      feature_truncation,non_coding_transcript_exon_variant   2063-?  -       -       -       -       -       IMPACT=HIGH;STRAND=-1;OverlapBP=188;OverlapPC=1.72
gnomAD_v2_BND_1_4       1:251749-258100 <BND>   ENSG00000228463 ENST00000450734 Transcript      downstream_gene_variant -       -       -       -       -       -       IMPACT=MODIFIER;DISTANCE=468;STRAND=-1
gnomAD_v2_BND_1_4       1:251749-258100 <BND>   ENSG00000228463 ENST00000634344 Transcript      downstream_gene_variant -       -       -       -       -       -       IMPACT=MODIFIER;DISTANCE=424;STRAND=-1

ref_panel_1kg_v1_BND_chr22_33_M2        22:22893781     ]chr22:22857058]G       ENSG00000211674 ENST00000390320 Transcript      feature_truncation,coding_sequence_variant      90      90      30      -       -       -       IMPACT=HIGH;STRAND=1;FLAGS=cds_start_NF,cds_end_NF;SYMBOL=IGLJ1;SYMBOL_SOURCE=HGNC;HGNC_ID=HGNC:5863;OverlapBP=1;OverlapPC=0.79
ref_panel_1kg_v1_BND_chr22_33_M2        22:22893781     ]chr22:22857058]G       ENSG00000211675 ENST00000390321 Transcript      upstream_gene_variant   -       -       -       -       -       -       IMPACT=MODIFIER;DISTANCE=1594;STRAND=1;FLAGS=cds_start_NF;SYMBOL=IGLC1;SYMBOL_SOURCE=HGNC;HGNC_ID=HGNC:5855
ref_panel_1kg_v1_BND_chr22_33_M2        22:22893781     ]chr22:22857058]G       ENSG00000226420 ENST00000432300 Transcript      intergenic_variant      -       -       -       -       -       -       IMPACT=MODIFIER;STRAND=1;SYMBOL=IGLV3-4;SYMBOL_SOURCE=HGNC;HGNC_ID=HGNC:5915
ref_panel_1kg_v1_BND_chr22_33_M2        22:22893781     ]chr22:22857058]G       ENSG00000253234 ENST00000522922 Transcript      intergenic_variant      -       -       -       -       -       -       IMPACT=MODIFIER;STRAND=1;SYMBOL=IGLV2-5;SYMBOL_SOURCE=HGNC;HGNC_ID=HGNC:5894
ref_panel_1kg_v1_BND_chr22_33_M2        22:22893781     ]chr22:22857058]G       ENSG00000254709 ENST00000526893 Transcript      feature_truncation,coding_sequence_variant      526     288     96      -       -       -       IMPACT=HIGH;STRAND=1;SYMBOL=IGLL5;SYMBOL_SOURCE=HGNC;HGNC_ID=HGNC:38476;OverlapBP=1;OverlapPC=0.01
ref_panel_1kg_v1_BND_chr22_33_M2        22:22893781     ]chr22:22857058]G       ENSG00000254709 ENST00000531372 Transcript      feature_truncation,intron_variant       -       -       -       -       -       -       IMPACT=HIGH;STRAND=1;SYMBOL=IGLL5;SYMBOL_SOURCE=HGNC;HGNC_ID=HGNC:38476;OverlapBP=1;OverlapPC=0.01
ref_panel_1kg_v1_BND_chr22_33_M2        22:22893781     ]chr22:22857058]G       ENSG00000254709 ENST00000532223 Transcript      feature_truncation,coding_sequence_variant      565     291     97      -       -       -       IMPACT=HIGH;STRAND=1;SYMBOL=IGLL5;SYMBOL_SOURCE=HGNC;HGNC_ID=HGNC:38476;OverlapBP=1;OverlapPC=0.01

BND_BRCA1       17:43044185     ]2:7938377]C    ENSG00000012048 ENST00000352993 Transcript      downstream_gene_variant -       -       -       -       -       -       IMPACT=MODIFIER;DISTANCE=110;STRAND=-1;SYMBOL=BRCA1;SYMBOL_SOURCE=HGNC;HGNC_ID=HGNC:1100
BND_BRCA1       17:43044185     ]2:7938377]C    ENSG00000012048 ENST00000357654 Transcript      downstream_gene_variant -       -       -       -       -       -       IMPACT=MODIFIER;DISTANCE=110;STRAND=-1;SYMBOL=BRCA1;SYMBOL_SOURCE=HGNC;HGNC_ID=HGNC:1100
BND_BRCA1       17:43044185     ]2:7938377]C    ENSG00000235665 ENST00000426969 Transcript      feature_truncation      -       -       -       -       -       -       IMPACT=HIGH;STRAND=-1;SYMBOL=LINC00298;SYMBOL_SOURCE=HGNC;HGNC_ID=HGNC:49257
BND_BRCA1       17:43044185     ]2:7938377]C    ENSG00000235665 ENST00000456681 Transcript      feature_truncation      -       -       -       -       -       -       IMPACT=HIGH;STRAND=-1;SYMBOL=LINC00298;SYMBOL_SOURCE=HGNC;HGNC_ID=HGNC:49257
BND_BRCA1       17:43044185     ]2:7938377]C    ENSG00000012048 ENST00000461221 Transcript      downstream_gene_variant -       -       -       -       -       -       IMPACT=MODIFIER;DISTANCE=1493;STRAND=-1;SYMBOL=BRCA1;SYMBOL_SOURCE=HGNC;HGNC_ID=HGNC:1100
BND_BRCA1       17:43044185     ]2:7938377]C    ENSG00000012048 ENST00000468300 Transcript      downstream_gene_variant -       -       -       -       -       -       IMPACT=MODIFIER;DISTANCE=620;STRAND=-1;SYMBOL=BRCA1;SYMBOL_SOURCE=HGNC;HGNC_ID=HGNC:1100
BND_BRCA1       17:43044185     ]2:7938377]C    ENSG00000012048 ENST00000471181 Transcript      downstream_gene_variant -       -       -       -       -       -       IMPACT=MODIFIER;DISTANCE=110;STRAND=-1;SYMBOL=BRCA1;SYMBOL_SOURCE=HGNC;HGNC_ID=HGNC:1100
BND_BRCA1       17:43044185     ]2:7938377]C    ENSG00000012048 ENST00000491747 Transcript      downstream_gene_variant -       -       -       -       -       -       IMPACT=MODIFIER;DISTANCE=1493;STRAND=-1;SYMBOL=BRCA1;SYMBOL_SOURCE=HGNC;HGNC_ID=HGNC:1100
BND_BRCA1       17:43044185     ]2:7938377]C    ENSG00000012048 ENST00000493795 Transcript      downstream_gene_variant -       -       -       -       -       -       IMPACT=MODIFIER;DISTANCE=1444;STRAND=-1;SYMBOL=BRCA1;SYMBOL_SOURCE=HGNC;HGNC_ID=HGNC:1100
BND_BRCA1       17:43044185     ]2:7938377]C    ENSG00000012048 ENST00000586385 Transcript      downstream_gene_variant -       -       -       -       -       -       IMPACT=MODIFIER;DISTANCE=1378;STRAND=-1;SYMBOL=BRCA1;SYMBOL_SOURCE=HGNC;HGNC_ID=HGNC:1100
BND_BRCA1       17:43044185     ]2:7938377]C    ENSG00000012048 ENST00000591534 Transcript      downstream_gene_variant -       -       -       -       -       -       IMPACT=MODIFIER;DISTANCE=1378;STRAND=-1;SYMBOL=BRCA1;SYMBOL_SOURCE=HGNC;HGNC_ID=HGNC:1100
BND_BRCA1       17:43044185     ]2:7938377]C    ENSG00000012048 ENST00000591849 Transcript      downstream_gene_variant -       -       -       -       -       -       IMPACT=MODIFIER;DISTANCE=1378;STRAND=-1;SYMBOL=BRCA1;SYMBOL_SOURCE=HGNC;HGNC_ID=HGNC:1100
BND_BRCA1       17:43044185     ]2:7938377]C    ENSG00000012048 ENST00000644379 Transcript      downstream_gene_variant -       -       -       -       -       -       IMPACT=MODIFIER;DISTANCE=901;STRAND=-1;FLAGS=cds_start_NF;SYMBOL=BRCA1;SYMBOL_SOURCE=HGNC;HGNC_ID=HGNC:1100
BND_BRCA1       17:43044185     ]2:7938377]C    ENSG00000286857 ENST00000655474 Transcript      feature_truncation      -       -       -       -       -       -       IMPACT=HIGH;STRAND=1
BND_BRCA1       17:43044185     ]2:7938377]C    ENSG00000235665 ENST00000663636 Transcript      feature_truncation      -       -       -       -       -       -       IMPACT=HIGH;STRAND=-1;SYMBOL=LINC00298;SYMBOL_SOURCE=HGNC;HGNC_ID=HGNC:49257
BND_BRCA1       17:43044185     ]2:7938377]C    ENSG00000235665 ENST00000664226 Transcript      intergenic_variant      -       -       -       -       -       -       IMPACT=MODIFIER;STRAND=-1;SYMBOL=LINC00298;SYMBOL_SOURCE=HGNC;HGNC_ID=HGNC:49257
```

## Potential improvements

- Return which breakend mate is associated with each feature consequence
- If possible, avoid returning `feature_truncation` without more information on which features are affected (UTR, exon, etc.)
- Add unit tests

### CODE REVIEW

**Review by likhitha-surapaneni — COMMENTED** [review]
LGTM. Just a minor change

**Comment by nuno-agostinho** [comment]
Needs to be backported to `release/110`.

File: modules/Bio/EnsEMBL/Variation/Utils/VariationEffect.pm [hunk 1]
```diff
@@ -288,10 +307,20 @@ sub feature_elongation {
 
 sub feature_truncation {
     my ($bvfoa, $feat, $bvfo, $bvf) = @_;
+    $bvf  ||= $bvfoa->base_variation_feature;
     $feat ||= $bvfoa->base_variation_feature_overlap->feature;
     
     return 0 if $bvfoa->isa('Bio::EnsEMBL::Variation::TranscriptVariationAllele');
     
+    if(breakend(@_)) {
+        my $chr;
```
[the remark this answers is not in the document]
Comment (nuno-agostinho) [reply modules/Bio/EnsEMBL/Variation/Utils/VariationEffect.pm]: No, removed the variable, thanks!

**Review by likhitha-surapaneni — COMMENTED** [review]
LGTM! Thanks @nuno-agostinho

**Comment by nuno-agostinho** [comment]
Merged to `release/111`, `postreleasefix/111`, `release/110` and `postreleasefix/110`.

**Pull request merged**